contracts/rosca.sol

```diff
@@ -1,266 +1,258 @@
 pragma solidity ^0.4.4;
 
-/*
-*  Possible missing functions
-* - integrity check? to perform check and make sure certain values are available and correct
-*
+/**
+*  Things still missing/needs fixing:
+*   in bid(), check if the ROSCA/rounds is started before allow bidding
+*   in startRound() , check if minimumParticipants requirement is met (what happens if it doesn't? should we move the startTime?)
+*   in withdraw() , add fee related logic
+*   in startRound() , lowestBid should be contributionSize * amount of people in ROSCA
+*   checkingContribution, check whether or not a member as contributed as least contributionSize at the end of Round
+*   in withdraw() , check if opt_destination provided is valid address
 */
-contract rosca {
-    address constant WETRUST = 0x0;
-
-    enum Period { Weekly, Monthly } // are we only allwing 2 options?
-
-    event LogParticipantRegistered(address user);
-    event LogContributionMade(address user, uint amount);
-    event LogNewLowestBid(uint bid,address winnerAddress);
-    event LogRoundFundsRelease(address winnerUniqName, uint amountInWei);
-
-
-    //added
-    event LogFundsWithdrawal(address user, uint amount,address destination);
-    event LogStartOfPeriod(uint currentRound);
-
-    //state variables
-    address foreman;
-    uint roundSum;
-    uint numRounds;
-    uint currentEpoch;
-    uint numEpochs;
-    int8 minParticipants;
-    uint roundPeriodInDays;
-    uint startTime;
-    int16 feeToService;
-    address feeAddress;
-    uint currentRound;  //currentRound will be set to 0 when ROSCA is created and will turn to one when the ROSCA actually start
-
-    /*
-        not sure about which structures to use yet
+contract ROSCA {
+  address constant WETRUST = 0x0;
+  uint constant MIN_CONTRIBUTION_SIZE = 1000000000000;
+  uint constant MAX_FEE_IN_THOUSANDTH = 20;
+
+  event LogParticipantRegistered(address user);
+  event LogContributionMade(address user, uint amount);
+  event LogNewLowestBid(uint bid,address winnerAddress);
+  event LogRoundFundsRelease(address winnerUniqName, uint amountInWei);
```

Comment:
1. s/LogRoundFundsRelease/LogRoundFundsReleased
2. s/winnerUniqName/winnerAddress (because we no longer have unique name strings for participant)